qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -156,6 +156,84 @@ def to_printer(self, printer, callback=None):
         self._widget.page().print(printer, callback)
 
 
+class _WebEngineSearchWrapHandler:
+
+    """QtWebEngine implementations related to wrapping when searching.
+
+    Attributes:
+        flag_wrap: An additional flag indicating whether the last search
+                   used wrapping.
+        _active_match: The 1-based index of the currently active match
+                       on the page.
+        _total_matches: The total number of search matches on the page.
+        _nowrap_available: Whether the functionality to prevent wrapping
+                           is available.
+    """
+
+    def __init__(self):
+        self._active_match = 0
+        self._total_matches = 0
+        self.flag_wrap = True
+        self._nowrap_available = False
+
+    def connect_signal(self, page):
+        """Connect to the findTextFinished signal of the page.
+
+        Args:
+            page: The QtWebEnginePage to connect to this handler.
+        """
+        if qtutils.version_check("5.14"):
+            page.findTextFinished.connect(self._store_match_data)
+            self._nowrap_available = True
+
+    def _store_match_data(self, result):
+        """Store information on the last match.
+
+        The information will be checked against when wrapping is turned off.
+
+        Args:
+            result: A FindTextResult passed by the findTextFinished signal.
+        """
+        self._active_match = result.activeMatch()
+        self._total_matches = result.numberOfMatches()
+        log.webview.debug("Active search match: {}/{}"
+                          .format(self._active_match, self._total_matches))
+
+    def reset_match_data(self):
+        """Reset match information.
+
+        Stale information could lead to next_result or prev_result misbehaving.
+        """
+        self._active_match = 0
+        self._total_matches = 0
+
+    def _message_wrap_prevented(self, top=True):
+        """Show a message informing the user that wrapping was prevented."""
+        if top:
+            message.info("Search hit TOP")
+        else:
+            message.info("Search hit BOTTOM")
+
+    def prevent_wrapping(self, going_up):
```

Comment:
```suggestion
    def prevent_wrapping(self, *, going_up):
```

so that the bool needs to be passed as keyword-argument.